library/hkdf.c

```diff
@@ -60,6 +60,9 @@ int mbedtls_hkdf_extract( const mbedtls_md_info_t *md,
 
     if( salt == NULL )
     {
+        if( salt_len != 0 ) {
+            return MBEDTLS_ERR_HKDF_BAD_INPUT_DATA;
+        }
         size_t hash_len;
```

Comment:
Please add the `if( salt_len != 0 ) {` block below the hash_len declaration to keep C89 compatibility.

```C
if( salt == NULL )
{
    size_t hash_len;

    if( salt_len != 0 ) {
        return MBEDTLS_ERR_HKDF_BAD_INPUT_DATA;
    }

    hash_len = mbedtls_md_get_size( md );
}
````

Reply:
done